scripts/gen_distro_docs.py

```diff
@@ -2,14 +2,43 @@
 
 import yaml
 import re
+import subprocess
 from pathlib import Path
 
 
 REPO_ROOT = Path(__file__).parent.parent
 
 
+def resolve_main_commit_hash(repo_url):
+    """Resolve the commit hash from the main branch of a git repository."""
+    try:
+        # Use git ls-remote to get the commit hash of the main branch
+        result = subprocess.run(
+            ["git", "ls-remote", repo_url, "main"],
+            capture_output=True,
+            text=True,
+            check=True,
+            timeout=10,
+        )
+        # Output format: <commit_hash>\trefs/heads/main
+        lines = result.stdout.strip().split("\n")
+        if lines and lines[0]:
+            commit_hash = lines[0].split()[0]
+            # Return full commit hash
+            return commit_hash
+        return None
+    except (subprocess.CalledProcessError, subprocess.TimeoutExpired, Exception) as e:
+        # If we can't resolve, return None to fall back to "main"
+        print(f"Warning: Could not resolve commit hash from main: {e}")
+        return None
+
+
 def extract_llama_stack_version():
```

Comment:
maybe add a type hint here since there's a specific return type mentioned in the docstring?